src/web/src/app/space/service.ts

```diff
@@ -124,40 +124,32 @@ export class SpaceService {
 
 		this._isChecked = true;
 		// TODO: remove next line
-		return of({} as SpaceServiceInfo).pipe(
+		//return of({} as SpaceServiceInfo).pipe(
 		// TODO: uncomment next line
-		// return this.http.get<SpaceServiceInfo>(this._apiUrl + '/info').pipe(
-			tap(res => this._isInitialized.next(res))
+		return this.http.get<SpaceServiceInfo>(this._apiUrl + '/info').pipe(
+			tap(res => this._isInitialized.next(res)),
+			src => this.requestWatcher.WatchFor(src),
 		);
 	}
 
 	isInitialized(): Observable<SpaceServiceInfo> {
 		return this._isInitialized.asObservable();
 	}
 
-	getList(limit: number, page: number, name: string = ''): Observable<PaginatorApi<Space>> {
-		const grouped = this.spaces.filter(space => space.name.includes(name));
-		const res: PaginatorApi<Space> = {
-			count: grouped.length,
-			results: grouped.slice(limit * page, limit * page + limit),
-		};
-		return of(res)
-			.pipe(delay(2000));
+	getList(limit: number, page: number, name: string = ''): Observable<Paging<Space>> {
+		return this.http.get<Paging<Space>>(this._apiUrl + "/list?start=" + limit*page + "&limit=" + limit).pipe(
+			src => this.requestWatcher.WatchFor(src)
+		);
 	}
 
-	getAvailableList(limit: number, page: number, name: string = '', userId: string): Observable<PaginatorApi<Space>> {
-		const grouped = this.spaces.filter(space => {
-			return space.requestsAllowed
-				&& space.name.includes(name)
-				&& !this.users.find(u => u.spaceId == space.id && u.userId == userId);
-		});
-
-		const res: PaginatorApi<Space> = {
-			count: grouped.length,
-			results: grouped.slice(limit * page, limit * page + limit),
-		};
-		return of(res)
-			.pipe(delay(2000));
+	getAvailableList(limit: number, page: number, name: string = ''): Observable<Paging<Space>> {
+		if (name != '')
+			return this.http.get<Paging<Space>>(this._apiUrl + "/available/list?spaceName=" + name + "&start=" + limit*page + "&limit=" + limit).pipe(
+			src => this.requestWatcher.WatchFor(src)
+		)
+		return this.http.get<Paging<Space>>(this._apiUrl + "/available/list?start=" + limit*page + "&limit=" + limit).pipe(
+			src => this.requestWatcher.WatchFor(src)
+		);
```

Comment:
Кажется тут можно чуть короче
```suggestion
		let url = this._apiUrl + "/available/list?start=" + limit*page + "&limit=" + limit;
		if (name)
			url += '&spaceName=' + name;
		return this.http.get<Paging<Space>>(url).pipe(
			src => this.requestWatcher.WatchFor(src)
		);
```